File: sandbox-runner/patterns/event-accumulator/go/shared.go
package main

import "time"

const (
	TaskQueue        = "accumulator-task-queue"
	WorkflowIDPrefix = "accumulator"
)

// maxAwaitTime is the sliding inactivity window. Use minutes (e.g. 30) in production.
var maxAwaitTime = 10 * time.Second

// OrderItem is a single event signal payload. ItemID is the deduplication key.
type OrderItem struct {
	OrderID string `json:"order_id"`
	ItemID  string `json:"item_id"`
	Name    string `json:"name"`
}

File: sandbox-runner/patterns/event-accumulator/java/src/main/java/Shared.java
import java.time.Duration;

public final class Shared {
    public static final String TASK_QUEUE = "accumulator-task-queue";
    public static final String WORKFLOW_ID_PREFIX = "accumulator";

    // Sliding inactivity window. Use minutes (e.g. Duration.ofMinutes(30)) in production.
    public static final Duration MAX_AWAIT_TIME = Duration.ofSeconds(10);

    public static class OrderItem {
        public String orderId;
        public String itemId;
        public String name;

        public OrderItem() {}

        public OrderItem(String orderId, String itemId, String name) {
            this.orderId = orderId;
            this.itemId = itemId;
            this.name = name;
        }

        @Override
        public String toString() {
            return name + " (" + itemId + ")";
        }
    }

    private Shared() {}
}
